src/Symfony/Component/DomCrawler/Crawler.php

```diff
@@ -501,16 +501,28 @@ public function parents()
     /**
      * Returns the children nodes of the current selection.
      *
+     * @param string|null $selector An optional CSS selector to filter children
+     *
      * @return self
      *
      * @throws \InvalidArgumentException When current node is empty
+     * @throws \RuntimeException         If the CssSelector Component is not available and $selector is provided
      */
-    public function children()
+    public function children(/* string $selector = null */)
```

Comment:
@nicolas-grekas shouldn't we also detect cases where `getclass($this) !== __CLASS__` (plus the special handling of common mock libraries) to trigger a deprecation warning if they don't have the argument in the child class ? Otherwise, we don't have a continous migration path.

Reply:
Looking at the source, we forgot such deprecations in many places.
Maybe merge as is (the continuous upgrade path would be just *not* adding this argument in v5)
and fix all the code base at once next?